crates/matrix-sdk/src/client/mod.rs

```diff
@@ -2240,6 +2240,28 @@ impl Client {
         // SAFETY: always initialized in the `Client` ctor.
         self.inner.event_cache.get().unwrap()
     }
+
+    /// Waits until an at least partially synced room is received, and returns
+    /// it.
+    ///
+    /// **Note: this function will loop endlessly until either it finds the room
+    /// or an externally set timeout happens.**
+    pub async fn await_room_remote_echo(&self, room_id: &RoomId) -> Room {
+        loop {
+            if let Some(room) = self.get_room(room_id) {
+                if room.is_state_partially_or_fully_synced() {
+                    debug!("Found just created room!");
+                    return room;
+                } else {
+                    warn!("Room wasn't partially synced, waiting for sync beat to try again");
+                }
+            } else {
+                warn!("Room wasn't found, waiting for sync beat to try again");
+            }
+            self.inner.sync_beat.listen().await;
+            debug!("New sync beat found");
```

Comment:
and this one we can remove, since the next loop iteration will immediately log something or return.